intern_program/intern_program_fall_2018_projects.md

```diff
@@ -34,3 +34,14 @@ This includes building a `.json` catalog of the existing datasets, integration a
 
 #### Skills Required
 Python programming, basic knowledge of html
+
+### Modeling time in the IRI Data Library (DL): new functionality needed
+
+#### Supervisor
+Rémi COUSIN, Climate Information Analyst
+
+#### Description
+The IRI DL allows for the online analysis of hundreds of variables by the scientific or practice community. A new batch of variables comes from so called subseasonal-to-seasonal (S2S) climate models outputs that are issued every day or at a lesser frequency, with daily lead times up to typically 40 days. The time dimensions of those outputs need to be manipulated, sampled, aggregated, to match the time dimension of other variables so that they can be compared (e.g. against other models, observations). The IRI DL Ingrid programming language can already achieve a lot to manipulate and transform time, however, new easy-to-use functions are needed to facilitate these manipulations for non-Ingrid experts. The intern will familiarize her/himself with Ingrid and develop new easy-to-use Ingrid functions to perform time analyses on S2S variables.
+
+#### Skills Required
+Willing to learn Ingrid programming language
```

Comment:
Can you tell me more about the Ingrid programming language? Is it based on another language or is it a full language? Is there a Wikipedia page?

Reply:
Hi! Ingrid is rather a full language even though it finds its roots in postscript. It's a stack-based language. You can find out a bit more for yourself visiting either of the 2 tutorials or the Function Documentation at this page: http://iridl.ldeo.columbia.edu/dochelp/index.html
(The Video Tutorial is still relevant too even though the UI has changed a bit).

Reply:
Ingrid looks very interesting! I guess my only concern is asking an intern to learn a new very specialized language that he or she may never use again, and will be difficult to explain on a CV. The interns are usually very interested in getting experience that will help them in terms of future employment. Of course there is probably no alternative here and maybe it will be a good learning experience. @rabernat may want to weigh in.

Reply:
That is very fair comment. No hard feeling if you drop my proposal. It's likely they never use it again unless they want to work for the IRI. If not, indeed, they can only put forward to recruiters that they were able to learn a new language and be useful with it.

This project is indeed code development. I could propose something more "applied", ie more data analysis on climate science topics, so that then, Ingrid would be a means rather than the finality, but that may not be what you seek either.

We are also trying to "plug" Ingrid and the Data Library with 3rd party more popular software and programming languages, or to build up the UI to keep the user away from actually programming in Ingrid on the website, and I am sure there would be ton of a work there, but I would have to think through that a bit more, or to leave that to another colleague.